File: 1646-kth-missing-positive-number/kth-missing-positive-number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        i, n = 0, 1
        l = len(arr)
        while k and i < l:
            if n < arr[i]:
                k -= 1
                n += 1
            else:
                i += 1
                n += 1
        return n + k - 1

File: 1646-kth-missing-positive-number/kth-missing-positive-number.c
int findKthPositive(int* arr, int arrSize, int k) {
    int i = 0, n = 1;

    while (k && i < arrSize) {
        if (n < arr[i])
            k -= 1;
        else
            i += 1;
        n += 1;
    }
    return n + k - 1;
}